drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -226,13 +253,81 @@ static int siwx917_scan(const struct device *dev, struct wifi_scan_params *z_sca
 static int siwx917_status(const struct device *dev, struct wifi_iface_status *status)
 {
 	struct siwx917_dev *sidev = dev->data;
+	sl_si91x_rsp_wireless_info_t info = {};
+	sl_wifi_rate_t mask = {0};
+	int ret;
 	int32_t rssi = -1;
 
+	__ASSERT(status, "status should not be NULL");
+
 	memset(status, 0, sizeof(*status));
 	status->state = sidev->state;
-	sl_wifi_get_signal_strength(SL_WIFI_CLIENT_INTERFACE, &rssi);
-	status->rssi = rssi;
-	return 0;
+
+	ret = sl_wifi_get_wireless_info(&info);
+	if (ret) {
+		LOG_ERR("Failed to get the wireless info: 0x%x", ret);
+		return -ENOTSUP;
```

Comment:
Is `ENOTSUP` the correct error? I believe this error is not expected whatever the context. I would use `EIO` in this case.